test/model/deepstate/test_issm.py

```diff
@@ -0,0 +1,269 @@
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import numpy as np
+import pandas as pd
+import mxnet as mx
+
+from gluonts.time_feature import (
+    DayOfWeek,
+    MonthOfYear,
+)
+from gluonts.model.deepstate.issm import (
+    LevelISSM,
+    LevelTrendISSM,
+    SeasonalityISSM,
+    CompositeISSM,
+    ZeroFeature,
+)
+
+
+def test_zero_feature():
+    zf = ZeroFeature()
+    feature = zf(pd.date_range("2020-01-01 21:00:00", periods=10, freq="H"))
+
+    assert isinstance(feature, np.ndarray)
+    assert feature.shape == (10,)
+    assert feature.dtype == np.float
+
+
+def test_level_issm_h():
+    issm = LevelISSM()
+
+    assert issm.latent_dim() == 1
+    assert issm.output_dim() == 1
+
+    time_features = issm.time_features()
+
+    assert len(time_features) == 1
+
+    time_indices = [
+        pd.date_range("2020-01-01 21:00:00", periods=10, freq="H"),
+        pd.date_range("2020-01-31 22:00:00", periods=10, freq="H"),
+    ]
+
+    features = mx.nd.array(
+        np.stack(
+            [
+                np.stack([f(time_index) for f in time_features], axis=-1)
+                for time_index in time_indices
+            ],
+            axis=0,
+        )
+    )
+
+    emission_coeff, transition_coeff, innovation_coeff = issm.get_issm_coeff(
+        features
+    )
+
+    assert emission_coeff.shape == (
+        2,
+        10,
+        issm.output_dim(),
+        issm.latent_dim(),
+    )
+    assert transition_coeff.shape == (
+        2,
+        10,
+        issm.latent_dim(),
+        issm.latent_dim(),
+    )
+    assert innovation_coeff.shape == (2, 10, issm.latent_dim())
+
+
+def test_level_trend_issm_h():
+    issm = LevelTrendISSM()
+
+    assert issm.latent_dim() == 2
+    assert issm.output_dim() == 1
+
+    time_features = issm.time_features()
+
+    assert len(time_features) == 1
+
+    time_indices = [
+        pd.date_range("2020-01-01 21:00:00", periods=10, freq="H"),
+        pd.date_range("2020-01-31 22:00:00", periods=10, freq="H"),
+    ]
+
+    features = mx.nd.array(
+        np.stack(
+            [
+                np.stack([f(time_index) for f in time_features], axis=-1)
+                for time_index in time_indices
+            ],
+            axis=0,
+        )
+    )
+
+    emission_coeff, transition_coeff, innovation_coeff = issm.get_issm_coeff(
+        features
+    )
+
+    assert emission_coeff.shape == (
+        2,
+        10,
+        issm.output_dim(),
+        issm.latent_dim(),
+    )
+    assert transition_coeff.shape == (
+        2,
+        10,
+        issm.latent_dim(),
+        issm.latent_dim(),
+    )
+    assert innovation_coeff.shape == (2, 10, issm.latent_dim())
+
+
+def test_seasonality_issm_h():
+    issm = SeasonalityISSM(
+        num_seasons=12, time_feature=MonthOfYear(normalized=False)
```

Comment:
The fact that `normalized=False` must be passed here is a bit annoying; as an alternative, we could either have alternative constructors that don't allow for normalization (eg `MonthOfYearIndex`) or just pass the feature *type* here, and construct the object internally.

Thoughts @rshyamsundar?

Reply:
yea.. passing only type and constructing the object internally as you did in the latest revision is better...